scripts/upgrade.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```

Comment:
This needs some comments or doc.  When do you run it?

Reply:
https://docs.google.com/document/d/1AqWfSBK8m8XFi23aQ5j3SLhO7AcFWFKG0SWRjUDT3EE/edit

This was more of a research task , I havent added documentation for it yet , the upgrades stuff is nowhere near being finalized . You could still use the above doc for reference